src/app/pages/things/details/things.details.component.ts

```diff
@@ -117,17 +120,26 @@ export class ThingsDetailsComponent implements OnInit, OnDestroy {
   getChannelMessages() {
     const messages: MainfluxMsg[] = [];
     this.connectedChans.forEach((chan, i) => {
-      this.messagesService.getMessages(chan.id, this.thing.key, this.thing.id).subscribe(
+      this.messagesService.getMessages(chan.id, this.thing.key, this.thing.id,
+        this.filters.subtopic, this.filters.offset, this.filters.limit, this.filters.from, this.filters.to).subscribe(
         (respMsg: any) => {
-          respMsg.messages.forEach((msg: MainfluxMsg) => messages.push(msg));
-          if (i === this.connectedChans.length - 1) {
-            this.messages = messages;
+          if (respMsg.messages) {
+            respMsg.messages.forEach((msg: MainfluxMsg) => messages.push(msg));
+            if (i === this.connectedChans.length - 1) {
+              this.messages = messages;
+            }
           }
         },
       );
     });
   }
 
+  onChangeDate(event) {
+    this.filters.from = event.from;
+    this.filters.to = event.to;
+    this.getChannelMessages();
+  }
```

Comment:
If I understand well, the problem with this code is that it still does not address the issue of pagination. I understand that it is difficult and that we do not have a pagination code infrastructure yet. I understand that it sort of works. However, it's confusing to the user, because user expects all messages from the given time range, and not just the first 20 messages. 

If you are unable to secure this delivery, be explicit about it, i.e. give user rather a possiblity to simply pick offset and limit (or something like that, like from message nr. x to message nr. y, where x < y).

There are many things that sort of work in the UI, so I suggest we rather solve these issues first. So, if you want to have this feature nevertheless, please mark it as experimental, because it does not work as intended.

Reply:
Pagination is related to the component. I set `messages-table` as experimental in Things and Channels details. We can remove it when this issue is fixed: https://github.com/mainflux/ui/issues/155